File: farmUnityClient/Resources/staticData.js
function StaticData() {
    var itemTypes = {
        currency: "currency",
        foundation: "foundation",
        building: "building",
        tool: "tool",
        seed: "seed"
    };

    var items = {
        softMoney: new Item({
            id: "softMoney",
            type: itemTypes.currency
        }),
        wateringCan: new Item({
            id: "wateringCan",
            type: itemTypes.tool,
            uncountable: true
        }),
        sickle: new Item({
            id: "sickle",
            type: itemTypes.tool,
            uncountable: true,
            use: function (target) {
                if (!target || target.id !== items.field.id) {
                    var targetString = target ? JSON.stringify(target) : undefined;
                    throw "Can't apply 'sickle', invalid target: " + targetString;
                }
                if (target.queue.length === 0) {
                    throw "Production queue is empty";
                }
                if (new Date().getTime() < target.endTime) {
                    throw "Field is not ready yet. It will be ready after " + (target.endTime - new Date().getTime()) + " milliseconds";
                }
                var reaped = target.queue.shift();
                target.endTime = undefined;
                target.currentProductionTimeLeft = undefined;
                bag.increaseCount(reaped.id, reaped.harvestValue);
                return utils.copy(target);
            }
        }),
        wheat: new Plant({
            id: "wheat",
            preparationTime: 10000,
            harvestValue: 3
        }),
        carrot: new Plant({
            id: "carrot",
            preparationTime: 50000,
            harvestValue: 3
        }),
        ground: new Item({
            id: "ground",
            type: itemTypes.foundation
        }),
        field: new Field({
            id: "field"
        }),
        well: new Item({
            id: "well",
            type: itemTypes.building
        })
    };

    this.getItemTypes = function () {
        return utils.copy(itemTypes);
    };

    this.getItems = function () {
        return utils.copy(items);
    };
}

var staticData = new StaticData();

File: farmBackend/src/main/resources/lua/staticData.lua
function StaticData()
    local itemTypes = {
        currency = "currency",
        foundation = "foundation",
        building = "building",
        tool = "tool",
        seed = "seed"
    }

    local items = {
        softMoney = Item({
            id = "softMoney",
            type = itemTypes.currency
        }),
        wateringCan = Item({
            id = "wateringCan",
            type = itemTypes.tool,
            uncountable = true
        }),
        sickle = Item({
            id = "sickle",
            type = itemTypes.tool,
            uncountable = true,
            use = function(target)
                if not target or target.id ~= items.field.id then
                    error("Can't apply 'sickle', invalid target: " .. target)
                end
                if #target.queue == 0 then
                    error("Production queue is empty")
                end
                if os.time() * 1000 < target.endTime then
                    error("Field is not ready yet. It will be ready after "..(target.endTime - os.time() * 1000).." milliseconds")
                end
                local reaped = table.remove(target.queue, 1)
                target.endTime = nil
                target.currentProductionTimeLeft = nil
                bag.increaseCount(reaped.id, reaped.harvestValue)
                return utils.copy(target)
            end
        }),
        wheat = Plant({
            id = "wheat",
            preparationTime = 10000,
            harvestValue = 3
        }),
        carrot = Plant({
            id = "carrot",
            preparationTime = 50000,
            harvestValue = 3
        }),
        ground = Item({
            id = "ground",
            type = itemTypes.foundation
        }),
        field = Field({}),
        well = Item({
            id = "well",
            type = itemTypes.foundation
        })
    }

    local getItemTypes = function()
        return utils.copy(itemTypes)
    end

    local getItems = function()
        return utils.copy(items)
    end

    return {
        getItemTypes = getItemTypes,
        getItems = getItems
    }
end

staticData = StaticData()
